webrender/src/frame.rs

```diff
@@ -757,6 +633,147 @@ impl Frame {
         context.builder.pop_layer();
     }
 
+    fn flatten_iframe<'a>(&mut self,
+                          pipeline_id: PipelineId,
+                          bounds: &Rect<f32>,
+                          context: &mut FlattenContext,
+                          current_scroll_layer_id: ScrollLayerId,
+                          layer_relative_transform: Matrix4D<f32>) {
+        context.pipeline_sizes.insert(pipeline_id, bounds.size);
+
+        let pipeline = match context.scene.pipeline_map.get(&pipeline_id) {
+            Some(pipeline) => pipeline,
+            None => return,
+        };
+
+        let display_list = context.scene.display_lists.get(&pipeline_id);
+        let display_list = match display_list {
+            Some(display_list) => display_list,
+            None => return,
+        };
+
+        let iframe_stacking_context = match display_list.starting_stacking_context() {
+            Some(stacking_context) => stacking_context,
+            None => {
+                warn!("Pipeline display list does not start with a stacking context.");
+                return;
+            }
+        };
+
+        self.pipeline_epoch_map.insert(pipeline_id, pipeline.epoch);
+
+        let iframe_rect = &Rect::new(Point2D::zero(), bounds.size);
+        let transform = layer_relative_transform.pre_translated(bounds.origin.x,
+                                                                bounds.origin.y,
+                                                                0.0);
+
+        let iframe_fixed_layer_id = ScrollLayerId::create_fixed(pipeline_id);
+        self.layers.insert(iframe_fixed_layer_id,
+                           Layer::new(&iframe_rect,
+                                      iframe_stacking_context.overflow.size,
+                                      &transform,
+                                      pipeline_id));
+
+        let iframe_scroll_layer_id = iframe_stacking_context.scroll_layer_id.unwrap();
+        self.layers.insert(iframe_scroll_layer_id,
+                           Layer::new(&iframe_rect,
```

Comment:
Okay. I've made Layer derive from Clone and done this. It made things quite a bit simpler.